### PR TITLE
Extend build system

### PR DESCRIPTION
**Pull request by aradi**
* Add toolchain used for staticly linked distributed binary.

* Centralize Scalapack detection to ensure consistent usage among
  external projects.

* Update install instructions

### CODE REVIEW

File: cmake/Modules/FindCustomScalapack.cmake [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  Customized ScaLAPACK library/libraries to use.  If no SCALAPACK library is
+  required (e.g. the linker automatically links it) set
+  ``SCALAPACK_LIBRARY="NONE"``. If not set or empty, it will use
+  ``find_package(scalapack)`` to find the scalapack library (relying on the
+  existence of a CMake export file for scalapack). Otherwise, the listed
+  libraries will be checked for existence (unless disabled in
+  ``SCALAPACK_DETECTION``) and the variable is overwritten to contain the
+  libraries with their with full path.
+
+  Note: On Ubuntu Focal (20.4 LTS) system, the installed CMake export file is broken.
```
Comment (bhourahine) [inline cmake/Modules/FindCustomScalapack.cmake:55]: Probably its older versions as well (the previous LTS also breaks it, so most likely almost all of them don't work).

**Review by bhourahine — APPROVED** [review]
Generally fine, but see comments.

**Review by bhourahine — COMMENTED** [review]
One issue on systems where the compiling nodes don't have git, but the head does so the submodules are downloaded. Then -DHYBRID_CONFIG_METHODS="Find" doesn't check the external/ directories.

File: sys/gnu-mkl-static.cmake [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+# ScaLAPACK -- only needed for MPI-parallel build
+#set(SCALAPACK_LIBRARY "scalapack-openmpi" CACHE STRING "Scalapack libraries to link")
+#set(SCALAPACK_LIBRARY_DIR "" CACHE STRING "Directories where Scalapack libraries can be found")
+
+# NOTE: The libraries below provide Pkg-Conf export files.  If your PKG_CONFIG_PATH environment
+# variable has been set up correctly (containing the paths to these libraries), no adjustment should
+# be necessary below.
+
+# PLUMED -- only needed when compiled with PLUMED support
```
[the remark this answers is not in the document]
Comment (aradi) [reply sys/gnu-mkl-static.cmake:105]: Well, it does assume that everything is there what is needed for the distributed binary...

**Comment by aradi** [comment]
@bhourahine I've fixed the submodule-problem, they should be used now if they are there, even if git is not available. Could you check it whether it works with your setup?

**Pull request merged**